kyaml/starlark/starlark_test.go

```diff
@@ -62,6 +63,83 @@ spec:
       - name: nginx
         # head comment
         image: nginx:1.8.1 # {"$ref": "#/definitions/io.k8s.cli.substitutions.image"}
+`,
+		},
+		{
+			name: "add_annotation_from_env",
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        # head comment
+        image: nginx:1.8.1 # {"$ref": "#/definitions/io.k8s.cli.substitutions.image"}
+`,
+			script: `
+def run(r):
+  for resource in r:
+    resource["metadata"]["annotations"]["foo"] = ctx.environment["ANNOTATION"]
+
+run(ctx.resource_list["items"])
+`,
+			env: map[string]string{"ANNOTATION": "annotation-value"},
+			expected: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  annotations:
+    foo: annotation-value
```

Comment:
Does this also update and delete existing annotations? If yes, can we add a test for that.

Reply:
no it doesn't